core/src/main/java/org/mule/runtime/core/internal/retry/policies/SimpleRetryPolicy.java

```diff
@@ -85,6 +82,11 @@ public SimpleRetryPolicy(long frequency, int retryCount) {
             fromExecutorService(new ConditionalExecutorServiceDecorator(retryScheduler, s -> isTransactionActive()));
 
         Mono<T> retryMono = from(publisher)
+            .doOnEach(e3 -> {
+              if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Retrying execution of event...");
```

Comment:
Is there any change chance to log the iteration number and how much left to get exhausted?

Reply:
I'm afraid I couldn't figure out how.